gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/HiveProxyQueryExecutor.java

```diff
@@ -0,0 +1,223 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.compliance;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.security.PrivilegedExceptionAction;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hive.jdbc.HiveConnection;
+import org.apache.thrift.TException;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+
+import edu.umd.cs.findbugs.annotations.SuppressWarnings;
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.util.HostUtils;
+
+
+/**
+ * This class is responsible for executing Hive queries using jdbc connection.
+ * This class can execute queries as hadoop proxy users by first authenticating hadoop super user.
+ *
+ * @author adsharma
+ */
+@Slf4j
+@SuppressWarnings
+public class HiveProxyQueryExecutor implements QueryExecutor, Closeable {
+  private static final String HIVE_SERVER2_PROXY_USER = "hive.server2.proxy.user=";
+  private static final String HIVE_JDBC_URL = ComplianceConfigurationKeys.COMPLIANCE_PREFIX + ".hive.jdbc.url";
+  private static final String DEFAULT = "default";
+  private static final String HIVE_SETTINGS = ComplianceConfigurationKeys.COMPLIANCE_PREFIX + ".hive.settings";
+  private static final Splitter SC_SPLITTER = Splitter.on(";").omitEmptyStrings().trimResults();
+  private Map<String, HiveConnection> connectionMap = new HashMap<>();
+  private Map<String, Statement> statementMap = new HashMap<>();
+  private State state;
+  private List<String> settings = new ArrayList<>();
+
+  public HiveProxyQueryExecutor(State state, List<String> proxies)
+      throws IOException {
+    try {
+      this.state = new State();
+      this.state.addAll(state.getProperties());
+      setHiveSettings(state);
+      setProxiedConnection(proxies);
+    } catch (InterruptedException | TException e) {
+      throw new IOException(e);
+    }
+  }
+
+  public HiveProxyQueryExecutor(State state)
+      throws IOException {
+    this.state = new State();
+    this.state.addAll(state.getProperties());
+    setHiveSettings(state);
+    try {
```

Comment:
You can create a static method `getProxiesFromState(State state)`, and then do `this(state, getProxiesFromState(state))` in this constructor.